test/encrypted_schema.test.js

```diff
@@ -0,0 +1,538 @@
+            name: { type: String, encrypt: { keyId: KEY_ID, algorithm } },
+            age: { type: Int32, encrypt: { keyId: KEY_ID, algorithm } },
+            name1: String,
+            age1: Int32
+          }, { encryptionType: 'csfle' });
+
+          const schema2 = originalSchema.omit(['name1', 'age1']);
+
+          assert.equal(schemaHasEncryptedProperty(schema2, ['name']), true);
+          assert.equal(schemaHasEncryptedProperty(schema2, ['age']), true);
+        });
+        it('Then the encryption type is the same as the original schema’s encryption type', function() {
+          const originalSchema = new Schema({
+            name: { type: String, encrypt: { keyId: KEY_ID, algorithm } },
+            age: { type: Int32, encrypt: { keyId: KEY_ID, algorithm } },
+            name1: String,
+            age1: Int32
+          }, { encryptionType: 'csfle' });
+
+          const schema2 = originalSchema.omit(['name1', 'age1']);
+
+          assert.equal(schema2.encryptionType(), 'csfle');
+        });
+      });
+
+      describe('When omit() is used with some unencrypted fields', function() {
+        it('Then the resultant schema has the encrypted fields of the original schema that were specified to omit()', function() {
+          const originalSchema = new Schema({
+            name: { type: String, encrypt: { keyId: KEY_ID, algorithm } },
+            age: { type: Int32, encrypt: { keyId: KEY_ID, algorithm } },
+            name1: String,
+            age1: Int32
+          }, { encryptionType: 'csfle' });
+
+          const schema2 = originalSchema.omit(['name', 'age1']);
+
+          assert.equal(schemaHasEncryptedProperty(schema2, ['name']), false);
+          assert.equal(schemaHasEncryptedProperty(schema2, ['age']), true);
+        });
+        it('Then the encryption type is the same as the original schema’s encryption type', function() {
+          const originalSchema = new Schema({
+            name: { type: String, encrypt: { keyId: KEY_ID, algorithm } },
+            age: { type: Int32, encrypt: { keyId: KEY_ID, algorithm } },
+            name1: String,
+            age1: Int32
+          }, { encryptionType: 'csfle' });
+
+          const schema2 = originalSchema.omit(['name', 'age1']);
+
+          assert.equal(schema2.encryptionType(), 'csfle');
+        });
+      });
+
+      describe('When omit() is used with some all the encrypted fields', function() {
+        it('Then the encryption type is the same as the original schema’s encryption type', function() {
+          const originalSchema = new Schema({
+            name: { type: String, encrypt: { keyId: KEY_ID, algorithm } },
+            age: { type: Int32, encrypt: { keyId: KEY_ID, algorithm } },
+            name1: String,
+            age1: Int32
+          }, { encryptionType: 'csfle' });
+
+          const schema2 = originalSchema.omit(['name', 'age']);
+
+          assert.equal(schema2.encryptionType(), 'csfle');
```

Comment:
If we only `pick` unencrypted fields, the resulting schema's `options.encryptionType` is `null`. 

With the current implementation, if we `omit` all encrypted fields, the resulting schema's `options.encryptionType` is not `null`. What's our reasoning for this?